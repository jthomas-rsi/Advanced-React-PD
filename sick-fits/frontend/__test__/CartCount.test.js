import { render, waitFor, screen } from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('CartCount testing suite', () => {
  it('Component renders correctly', () => {
    render(<CartCount count={10} />);
  });

  it('Component matches snapshot', () => {
    const { container } = render(<CartCount count={41} />);
    expect(container).toMatchSnapshot();
  });

  // test if component changes when props are changed
  it('Component updates via props', async () => {
    const { container, rerender } = render(<CartCount count={26} />);
    expect(container).toHaveTextContent('26');
    rerender(<CartCount count={30} />);
    // wait 400 ms for the component to update
    await wait(400);
    expect(container).toHaveTextContent('30');
  });
  it('Component updates via props 2', async () => {
    const { container, rerender } = render(<CartCount count={26} />);
    expect(container).toHaveTextContent('26');
    rerender(<CartCount count={30} />);
    await waitFor(() => {
      expect(container).toHaveTextContent('30');
    });
  });
  it('Component updates via props 3', async () => {
    const { rerender, debug } = render(<CartCount count={26} />);
    const count1 = screen.findByText('26');
    expect(count1).toBeTruthy();
    rerender(<CartCount count={30} />);
    await wait(400);
    const count2 = screen.findByText('30');
    expect(count2).toBeTruthy();
  });
});
