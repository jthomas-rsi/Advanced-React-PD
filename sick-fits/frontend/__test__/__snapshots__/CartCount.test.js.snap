// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartCount testing suite Component matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 kHAPbE"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 gRrnjj count"
      >
        41
      </div>
    </div>
  </span>
</div>
`;
